packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -574,10 +574,15 @@ export default createRule<Options, MessageId>({
         node.type === AST_NODE_TYPES.MemberExpression
           ? !isNullableOriginFromPrev(node)
           : true;
+      const unionParts = unionTypeParts(type);
+      const possiblyVoid = unionParts.some(
+        part => part.flags & ts.TypeFlags.Void,
+      );
       return (
         isTypeAnyType(type) ||
         isTypeUnknownType(type) ||
-        (isNullableType(type, { allowUndefined: true }) && isOwnNullable)
+        (isNullableType(type, { allowUndefined: true }) && isOwnNullable) ||
+        (possiblyVoid && isOwnNullable)
```

Comment:
The `&& isOwnNullable` is duplicated here. Let's simplify:

```ts
(isOwnNullable && (isNullableType(type, { allowUndefined: true }) || possiblyVoid))
```

Reply:
Updated